src/main/java/com/sap/broker/budgie/configuration/behavior/BehaviorEngine.java

```diff
@@ -0,0 +1,155 @@
+package com.sap.broker.budgie.configuration.behavior;
+
+import com.sap.broker.budgie.configuration.ApplicationConfiguration;
+import com.sap.broker.budgie.domain.Plan;
+import com.sap.broker.budgie.domain.Service;
+import com.sap.broker.budgie.domain.ServiceInstance;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiPredicate;
+import java.util.stream.Stream;
+
+@Component
+public class BehaviorEngine {
```

Comment:
*Engine... do not go in danger waters with names - be careful what names you put to your classes- try to be as much as possible explicit but do not make it too abstract. From this naming I could assume that this class is doing some heavy processing and complex analysis ...

Reply:
Does `ConfigurationManager` sound good?

Reply:
It Manager is the same as the Engine :D

Reply:
:D do you have any name in mind?